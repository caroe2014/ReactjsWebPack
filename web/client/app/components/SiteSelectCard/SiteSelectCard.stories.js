// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import { storiesOf } from '@storybook/react';
import SiteSelectCard from '.';

storiesOf('SiteSelectCard', module)
  .add('AvailabeAtDay', () => {
    const siteItem = {
      image: 'http://eatstreet.co.id/wp-content/uploads/2016/11/cropped-MASTER-FINAL_LOGO-OVAL-EAT-STREET-1.png',
      name: 'Eat Street',
      availableAt: {
        opens: '9:00am',
        closes: '5:00pm'
      },
      address: ['1000, windward concourse, Sometown', 'Alpharetta,GA, USA - 500100']
    };
    return (
      <SiteSelectCard keyProps={siteItem} />
    );
  })
  .add('AvailabeAtNight', () => {
    const siteItem = {
      image: 'http://eatstreet.co.id/wp-content/uploads/2016/11/cropped-MASTER-FINAL_LOGO-OVAL-EAT-STREET-1.png',
      name: 'Eat Street',
      availableAt: {
        opens: '5:00pm',
        closes: '9:00am'
      },
      address: ['1000, windward concourse', 'alpharetta,GA, USA']
    };
    return (
      <SiteSelectCard keyProps={siteItem} />
    );
  })
  .add('WithoutAddress', () => {
    const siteItem = {
      image: 'http://eatstreet.co.id/wp-content/uploads/2016/11/cropped-MASTER-FINAL_LOGO-OVAL-EAT-STREET-1.png',
      name: 'Eat Street',
      availableAt: {
        opens: '5:00pm',
        closes: '9:00am'
      }
    };
    return (
      <SiteSelectCard keyProps={siteItem} />
    );
  })
  .add('Without Image', () => {
    const siteItem = {
      name: 'Eat Street',
      availableAt: {
        opens: '9:00am',
        closes: '5:00pm'
      },
      address: ['1000, windward concourse', 'alpharetta,GA, USA']
    };
    return (
      <SiteSelectCard keyProps={siteItem} />
    );
  });
