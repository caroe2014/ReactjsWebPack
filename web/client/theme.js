// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import { css, injectGlobal } from 'styled-components';

const theme = {
  colors: {
    black: '#000000',
    navigationMenu: '#48a942',
    navigationMenuHover: 'cornflowerblue',
    primary: '#347CFF',
    primaryButton: '#347CFF',
    loginField: '#AAA7AC',
    placeHolder: '#909090',
    loginBoxColor: '#434149',
    open: '#48a942',
    close: 'red',
    footer: '#48a942',
    textGrey: '#505050',
    paymentSuccessTitle: '#505050',
    inputBottomBorder: 'lightgray',
    backgroundGrey: '#E6E6E6',
    text: 'black',
    primaryText: '#383838',
    secondaryText: '#a6a6a6',
    hint: '#a6a6a6',
    light: 'white',
    error: '#cc0000',
    tile: '#505050',
    addToCart: '#347CFF',
    itemTile: '#3C474C0',
    darkGrey: '#464646',
    itemBorder: 'lightgrey',
    addOnTab: '#347CFF',
    validationError: '#cc0000',
    loginBackground: '#303030',
    loginFormBackground: '#484848',
    disableBackground: '#E4E4E4',
    addonBg: '#EEEEEE',
    glutenFree: '#8d7356',
    spicy: '#ed1c24',
    vegan: '#0d4a22',
    vegetarian: '#0d4a22',
    new: '#ff6000',
    healthy: '#359919',
    contextBackground: '#f0f0f3',
    loader: '#ababab',
    disableButton: '#D1D1D1'
  },
  fontFamily: 'Roboto',
  receiptFontFamily: 'Roboto',
  fontSize: {
    xl: '2.4rem',
    lg: '1.8rem',
    md: '1.3rem',
    nm: '1rem',
    sm: '0.75rem'
  },
  fontSizeFromProps: (props) => {
    if (props.xl) return props.theme.fontSize.xl;
    if (props.lg) return props.theme.fontSize.lg;
    if (props.md) return props.theme.fontSize.md;
    if (props.nm) return props.theme.fontSize.nm;
    if (props.sm) return props.theme.fontSize.sm;
    return 'inherit';
  },
  spacing: {
    xl: '8em',
    lg: '4em',
    md: '2em',
    nm: '1em',
    sm: '0.5em',
    none: '0em'
  },
  mediaBreakpoints: {
    mobile: (...args) => css`@media (max-width: 32em) {${css(...args)}}`,
    tablet: (...args) => css`@media (max-width: 72em) {${css(...args)}}`,
    desktop: (...args) => css`@media (min-width: 72em) {${css(...args)}}`
  },
  shadow: '#cfcfcf 0.1px 0.1px 3px 0px',
  nav: {
    height: '70px'
  }
};

injectGlobal`
  .agilysys-icon-new {
    color: ${theme.colors.new}
  }
  .agilysys-icon-healthy {
    color: ${theme.colors.healthy}
  }
  .agilysys-icon-vegetarian {
    color: ${theme.colors.vegetarian}
  }
  .agilysys-icon-vegan {
    color: ${theme.colors.vegan}
  }
  .agilysys-icon-spicy {
    color: ${theme.colors.spicy}
  }
  .agilysys-icon-gluten-free {
    color: ${theme.colors.glutenFree}
  }
  @media print {
    @-moz-document url-prefix() {
      .appContainer, .appContainer>div, .appContainer .parent, .appContainer .BottomContainer {
        display: block !important;
      }
    }
  }
`;

export default theme;
