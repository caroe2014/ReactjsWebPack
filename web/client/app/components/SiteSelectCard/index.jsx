/* eslint-disable no-mixed-operators */
/* eslint-disable max-len */
// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React, { Component } from 'react';
import styled, { ThemeProvider } from 'styled-components';
import theme from 'web/client/theme';
import PropTypes from 'prop-types';
import { Flex, Text, Button } from 'rebass';
import Tile from 'web/client/app/components/Tile';
import i18n from 'web/client/i18n';
import { Trans } from 'react-i18next';
import { getDeliveryOptions } from 'web/client/app/utils/LocationUtils';

const Container = styled(Flex)`
 align-items: flex-start;
 flex-wrap: wrap;
 flex-direction: column;
 justify-content: center;
 margin: 0;
 padding: 0;
 max-width: 570px;
 width:100%;
 height: auto;
 &>div {
  height: 100%;
 }
 .top-container {
  ${props => props.theme.mediaBreakpoints.mobile`
    padding-bottom: 50px;
  `};
 }
`;

const TextContainer = styled(Text)`
 font-size: ${props => props.theme.fontSize.nm};
 margin-top: 2px;
 word-break: break-all;
 width: 100%;
 &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const OpenTextContainer = styled(Text)`
 font-size: 14px;
 margin-top: 16px;
 z-index: 10;
`;

const SiteStatusContainer = styled(Text)`
 font-size: 14px;
 z-index: 10;
`;

const AddressContainer = styled(Flex)`
  flex-grow: 1
  margin-top: 5px;
  color: ${props => props.theme.colors.secondaryTextColor};
  width: 100%;
`;

const Closed = styled(Text)`
  padding-top: 0.5em;
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const ClosedAsapMessage = styled(Text)`
  padding-top: 0.5em;
  padding-bottom: 0.5em;
  font-weight: 500;
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const DeliveryContainer = styled(Flex)`
  height: 35px;
  align-items: center;
  width: ${props => props.optionLength > 1 ? '100%' : '25%'};
  ${props => props.theme.mediaBreakpoints.mobile`
    height: 50px;
    position: absolute;
    left: 0;
    bottom: 0;
    padding: 0px 10px;
    width: ${props => props.optionLength > 1 ? '100%' : 'auto'};
    ${props => props.optionLength === 1 && `
    button {
      padding: 5px 15px;
    }`}
  `};
  &>:not(:last-child) {
    margin-right: 10px;
  }
`;

const DeliveryButton = styled(Button)`
  background: ${props => props.theme.colors.light};
  border: 0.5px solid ${props => props.theme.colors.buttonControlColor};
  color: ${props => props.theme.colors.buttonControlColor};
  cursor: pointer;
  font-weight: 500;
  font-size: ${props => props.theme.fontSize.nm};
  float: right;
  height: 30px;
  min-height: 30px;
  padding: 0px 0px 0px;
  flex: 1 1 auto;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  text-transform: uppercase;
  &:hover {
    box-shadow: 0px 0px 10px 2px lightgrey;
    -webkit-box-shadow: 0px 0px 10px 2px lightgrey;
    -moz-box-shadow: 0px 0px 10px 2px lightgrey;
  }
  text-align: center;
  border-radius: 5px;
  outline: none;
  &:focus {
    outline: none;
    box-shadow: none;
  }
  ${props => props.theme.mediaBreakpoints.mobile` padding: 2px 0px 0px;`};
`;

const openIcon = {
  color: theme.colors.open,
  fontSize: '16px',
  marginRight: '4px'
};

const closeIcon = {
  color: theme.colors.close,
  fontSize: '16px',
  marginRight: '4px'
};

const titleStyle = {
  fontSize: '18px'
};

class SiteSelectCard extends Component {

  constructor (props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
    const { keyProps } = this.props;
    this.deliveryOptions = getDeliveryOptions({ deliveryDestination: keyProps.deliveryDestination, dineInConfig: keyProps.dineInConfig, pickUpConfig: keyProps.pickUpConfig }); // eslint-disable-line max-len
  }
  handleClick (e) {
    if (this.props.keyProps.availableAt.availableNow && this.props.selectSite) {
      this.props.selectSite(this.props.keyProps);
    }
    this.props.setDeliveryOption(this.deliveryOptions.find(data => data.id === e.target.value));
  }
  render () {
    const { keyProps, ariaLabel, tabIndex, apptheme, isScheduleOrderEnabled, showOperationTimes, isAsapOrder } = this.props;
    const storeAvailabeNow = !isAsapOrder || (isAsapOrder && keyProps.storeAvailabeNow);
    const isEnabled = keyProps.availableAt.availableNow && keyProps.availableAt.conceptsAvailableNow && storeAvailabeNow;
    return (
      <ThemeProvider className='site-card' theme={theme}>
        <Container className='site-card-container'
          open={keyProps.availableAt.availableNow && keyProps.availableAt.conceptsAvailableNow &&
            storeAvailabeNow}>
          <Tile
            apptheme={apptheme}
            classContext={`tile site_${keyProps.id}`}
            title={keyProps.name}
            image={keyProps.image}
            imageresponsive='true'
            isEnabled={isEnabled}
            titleStyle={titleStyle}
            closedoverlay='true'
            ariaLabel={ariaLabel}
            tabIndex={tabIndex}
          >
            <AddressContainer
              className='address-text'
              flexDirection='column'
              alignItems='flex-start'
            >
              {keyProps.address && keyProps.address.map((item, index) => (
                <React.Fragment key={item.toString() + keyProps.id}>
                  <TextContainer aria-hidden={item === ' ' || item === '  '}
                    aria-label={item} tabIndex={(item === ' ' || item === '  ' || !isEnabled) ? -1 : 0}
                    className={`address-line-${index + 1}`}>{item}</TextContainer>
                </React.Fragment>
              ))}
              <OpenTextContainer className='opens-text custom-time'
                role={showOperationTimes && keyProps.availableAt.opens && 'img'}
                aria-label={showOperationTimes && keyProps.availableAt.opens && i18n.t('SITE_SELECT_OPEN_TIME',
                  { siteOpen: keyProps.availableAt.opens, siteClose: keyProps.availableAt.closes })}
                tabIndex={isEnabled ? 0 : -1}
              >
                {showOperationTimes && keyProps.availableAt.opens && <React.Fragment>
                  <i className='fa fa-clock-o'
                    alt=''
                    style={keyProps.availableAt.availableNow &&
                      keyProps.availableAt.conceptsAvailableNow && storeAvailabeNow ? openIcon : closeIcon}/>
                  {
                    i18n.t('SITE_SELECT_OPEN_TIME',
                      { siteOpen: keyProps.availableAt.opens, siteClose: keyProps.availableAt.closes })
                  }
                </React.Fragment>}
              </OpenTextContainer>
              <SiteStatusContainer>
                {keyProps.availableAt.availableNow &&
                  keyProps.availableAt.conceptsAvailableNow && isAsapOrder && !keyProps.storeAvailabeNow &&
                  <ClosedAsapMessage className='closes-asap' tabIndex={0} aria-label={i18n.t('CLOSED_ASAP')}><Trans i18nKey='CLOSED_ASAP' /></ClosedAsapMessage>
                }
                {(!isScheduleOrderEnabled || isAsapOrder) &&
                <React.Fragment>
                  {!keyProps.availableAt.availableNow &&
                    <Closed className='closes-text' tabIndex={0} aria-label={i18n.t('SITE_SELECT_CLOSED')}><Trans i18nKey='SITE_SELECT_CLOSED' /></Closed>
                  }
                  {keyProps.availableAt.availableNow && !keyProps.availableAt.conceptsAvailableNow &&
                    <Closed className='closes-text' tabIndex={0} aria-label={i18n.t('SITE_SELECT_OPEN_LATER')}><Trans i18nKey='SITE_SELECT_OPEN_LATER' /></Closed>
                  }
                </React.Fragment>
                }
              </SiteStatusContainer>
            </AddressContainer>
            {this.deliveryOptions.length > 0 &&
              <DeliveryContainer
                className='delivery-container'
                optionLength={this.deliveryOptions.length}>
                {this.deliveryOptions.map((data, index) =>
                  <DeliveryButton
                    className='delivery-button'
                    key={`${data.id}-${index}`}
                    children={data.displayText}
                    tabIndex={isEnabled ? 0 : -1}
                    value={data.id}
                    onClick={this.handleClick} />
                )}
              </DeliveryContainer>
            }
          </Tile>
        </Container>
      </ThemeProvider>
    );
  }
}

SiteSelectCard.propTypes = {
  keyProps: PropTypes.shape({
    id: PropTypes.string.isRequired,
    image: PropTypes.string,
    name: PropTypes.string.isRequired,
    availableAt: PropTypes.shape({
      opens: PropTypes.string.isRequired,
      closes: PropTypes.string.isRequired,
      availableNow: PropTypes.bool.isRequired,
      closingIn: PropTypes.number.isRequired
    }).isRequired,
    address: PropTypes.arrayOf(PropTypes.string.isRequired)
  }).isRequired
};
export default SiteSelectCard;
