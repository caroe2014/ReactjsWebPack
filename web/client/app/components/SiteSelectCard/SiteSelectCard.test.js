// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import {
  shallow,
  mount,
  configure
} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import {
  ThemeProvider
} from 'styled-components';
import theme from 'web/client/theme';

import SiteSelectCard from '.';

configure({
  adapter: new Adapter()
});

jest.mock('web/client/i18n', () => ({
  t: (key) => {
    return key;
  }
}));
jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate HoC receive the t function as a prop
  withTranslation: () => Component => {
    Component.defaultProps = { ...Component.defaultProps, t: (key) => key };
    return Component;
  },
}));

jest.mock('react-i18next', () => ({
  Trans: 'div'
}));

describe('SiteSelectCard', () => {
  let wrapper;
  let mockProps = {
    id: '1001',
    image: 'http://eatstreet.co.id/wp-content/uploads/2016/11/cropped-MASTER-FINAL_LOGO-OVAL-EAT-STREET-1.png',
    name: 'Eat Street',
    availableAt: {
      opens: '9:00am',
      closes: '6:00pm',
      availableNow: true,
      closingIn: 10,
      conceptsAvailableNow: true
    },
    address: ['1000, windward concourse', 'alpharetta,GA, USA']
  };

  beforeEach(() => {
    wrapper = shallow(<SiteSelectCard keyProps={mockProps} />
    );
  });

  it('does a snapshot test', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('should render without throwing an error on mount', () => {
    expect(mount(<ThemeProvider theme={theme}><SiteSelectCard keyProps={mockProps} showOperationTimes /></ThemeProvider>).exists()).toBe(true);
  });

  // it('checks whether image prop is recevied', () => {
  //   expect(wrapper.find('Tile').dive().find('.image').props().src).toBe(mockProps.image);
  // });

  describe('when Image prop is undefined', () => {
    beforeEach(() => {
      mockProps.image = '';
      mockProps.availableAt.conceptsAvailableNow = false;
    });
    it('checks whether image prop is not recevied', () => {
      const imageWrapper = shallow(< SiteSelectCard keyProps={mockProps}
      />);
      expect(imageWrapper.find('.image').length).toBe(0);
    });
    it('should render without throwing an error on mount', () => {
      expect(mount(<ThemeProvider theme={theme}><SiteSelectCard keyProps={mockProps} showOperationTimes /></ThemeProvider>).exists()).toBe(true);
    });
    it('should render without throwing an error on mount', () => {
      mockProps.availableAt.closingIn = 1;
      expect(mount(<ThemeProvider theme={theme}><SiteSelectCard keyProps={mockProps} showOperationTimes /></ThemeProvider>).exists()).toBe(true);
    });
  });

  it('show closed now text when store is closed', () => {
    mockProps.availableAt.availableNow = false;
    const closeWrapper = mount(<ThemeProvider theme={theme}><SiteSelectCard keyProps={mockProps} showOperationTimes /></ThemeProvider>);
    expect(closeWrapper.exists()).toBe(true);
  });

});
