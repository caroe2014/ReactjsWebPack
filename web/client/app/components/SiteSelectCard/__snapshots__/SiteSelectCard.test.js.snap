// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteSelectCard does a snapshot test 1`] = `
<ThemeProvider
  className="site-card"
  theme={
    Object {
      "colors": Object {
        "addOnTab": "#347CFF",
        "addToCart": "#347CFF",
        "addonBg": "#EEEEEE",
        "backgroundGrey": "#E6E6E6",
        "black": "#000000",
        "close": "red",
        "contextBackground": "#f0f0f3",
        "darkGrey": "#464646",
        "disableBackground": "#E4E4E4",
        "disableButton": "#D1D1D1",
        "error": "#cc0000",
        "footer": "#48a942",
        "glutenFree": "#8d7356",
        "healthy": "#359919",
        "hint": "#a6a6a6",
        "inputBottomBorder": "lightgray",
        "itemBorder": "lightgrey",
        "itemTile": "#3C474C0",
        "light": "white",
        "loader": "#ababab",
        "loginBackground": "#303030",
        "loginBoxColor": "#434149",
        "loginField": "#AAA7AC",
        "loginFormBackground": "#484848",
        "navigationMenu": "#48a942",
        "navigationMenuHover": "cornflowerblue",
        "new": "#ff6000",
        "open": "#48a942",
        "paymentSuccessTitle": "#505050",
        "placeHolder": "#909090",
        "primary": "#347CFF",
        "primaryButton": "#347CFF",
        "primaryText": "#383838",
        "secondaryText": "#a6a6a6",
        "spicy": "#ed1c24",
        "text": "black",
        "textGrey": "#505050",
        "tile": "#505050",
        "validationError": "#cc0000",
        "vegan": "#0d4a22",
        "vegetarian": "#0d4a22",
      },
      "fontFamily": "Roboto",
      "fontSize": Object {
        "lg": "1.8rem",
        "md": "1.3rem",
        "nm": "1rem",
        "sm": "0.75rem",
        "xl": "2.4rem",
      },
      "fontSizeFromProps": [Function],
      "mediaBreakpoints": Object {
        "desktop": [Function],
        "mobile": [Function],
        "tablet": [Function],
      },
      "nav": Object {
        "height": "70px",
      },
      "receiptFontFamily": "Roboto",
      "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
      "spacing": Object {
        "lg": "4em",
        "md": "2em",
        "nm": "1em",
        "none": "0em",
        "sm": "0.5em",
        "xl": "8em",
      },
    }
  }
>
  <Styled(Flex)
    className="site-card-container"
    open={true}
  >
    <Tile
      classContext="tile site_1001"
      closedoverlay="true"
      hasBorder={true}
      image="http://eatstreet.co.id/wp-content/uploads/2016/11/cropped-MASTER-FINAL_LOGO-OVAL-EAT-STREET-1.png"
      imageresponsive="true"
      isEnabled={true}
      title="Eat Street"
      titleStyle={
        Object {
          "fontSize": "18px",
        }
      }
    >
      <Styled(Flex)
        alignItems="flex-start"
        className="address-text"
        flexDirection="column"
      >
        <Styled(Text)
          aria-hidden={false}
          aria-label="1000, windward concourse"
          className="address-line-1"
          tabIndex={0}
        >
          1000, windward concourse
        </Styled(Text)>
        <Styled(Text)
          aria-hidden={false}
          aria-label="alpharetta,GA, USA"
          className="address-line-2"
          tabIndex={0}
        >
          alpharetta,GA, USA
        </Styled(Text)>
        <Styled(Text)
          className="opens-text custom-time"
          tabIndex={0}
        />
        <Styled(Text) />
      </Styled(Flex)>
      <Styled(Flex)
        className="delivery-container"
        optionLength={1}
      >
        <Styled(Button)
          className="delivery-button"
          key="dinein-0"
          onClick={[Function]}
          tabIndex={0}
          value="dinein"
        >
          DELIVERY_DINEIN_DISPLAY_TEXT
        </Styled(Button)>
      </Styled(Flex)>
    </Tile>
  </Styled(Flex)>
</ThemeProvider>
`;
